### ISSUE TITLE
[mpir] build failure

### ISSUE THREAD

**Issue by cenit** [microsoft/vcpkg#12708]
**Host Environment**
- OS: Ubuntu Linux 20.04
- Compiler: GCC 9.3

**To Reproduce**
`./vcpkg install mpir`

**Additional context**
after #12677 (removing autoconfig) mpir does not build anymore on Linux
@strega-nil

**Comment by NancyLi1013** [comment]
Hi @cenit 
Thanks for posting this issue.

@strega-nil 
It seems that `mpir `cannot build without `autoconfig` on Linux. Could you please help take a look?

```
../libtool: 1581: ../libtool: preserve_args+= --tag CC: not found
../libtool: 1: eval: base_compile+= gcc: not found
../libtool: 1: eval: base_compile+= -DHAVE_CONFIG_H: not found
../libtool: 1: eval: base_compile+= -I.: not found
../libtool: 1581: ../libtool: preserve_args+= --tag CC: not found
../libtool: 1: eval: base_compile+= gcc: not found
../libtool: 1: eval: base_compile+= -DHAVE_CONFIG_H: not found
../libtool: 1: eval: base_compile+= -I.: not found
../libtool: 1: eval: ../libtool: 1: eval: base_compile+= -I.././../src/3.0.0-a3f571144a.clean/mpn: not found
base_compile+= -I.././../src/3.0.0-a3f571144a.clean/mpn: not found
../libtool: 1: eval: ../libtool: 1: eval: base_compile+= -I..: not foundbase_compile+= -I..: not found

../libtool: 1: eval: ../libtool: 1: eval: base_compile+= -D__GMP_WITHIN_GMP: not foundbase_compile+= -D__GMP_WITHIN_GMP: not found

../libtool: 1: eval: base_compile+= -I.././../src/3.0.0-a3f571144a.clean: not found
../libtool: 1: eval: base_compile+= -DOPERATION_add: not found
../libtool: 1: eval: base_compile+= -Wno-implicit-function-declaration: not found
../libtool: 1: eval: base_compile+= -fPIC: not found
../libtool: 1: eval: base_compile+= -c: not found
libtool: compile: you must specify a compilation command
libtool: compile: Try `libtool --help --mode=compile' for more information.
../libtool: 1: eval: base_compile+= -I.././../src/3.0.0-a3f571144a.clean: not found
../libtool: 1581: ../libtool: preserve_args+= --tag CC: not found
../libtool: 1: eval: base_compile+= -DOPERATION_dummy1: not found
../libtool: 1: eval: base_compile+= -Wno-implicit-function-declaration: not found
../libtool: 1: eval: base_compile+= -fPIC: not found
../libtool: 1: eval: base_compile+= -c: not found
libtool: compile: you must specify a compilation command
libtool: compile: Try `libtool --help --mode=compile' for more information.
make[2]: *** [add.lo] Error 1
make[2]: *** Waiting for unfinished jobs....
make[2]: *** [dummy1.lo] Error 1
../libtool: 1: eval: base_compile+= gcc: not found
../libtool: 1: eval: base_compile+= -DHAVE_CONFIG_H: not found
../libtool: 1: eval: base_compile+= -I.: not found
../libtool: 1: eval: base_compile+= -I.././../src/3.0.0-a3f571144a.clean/mpn: not found
../libtool: 1: eval: base_compile+= -I..: not found
../libtool: 1: eval: base_compile+= -D__GMP_WITHIN_GMP: not found
../libtool: 1: eval: base_compile+= -I.././../src/3.0.0-a3f571144a.clean: not found
../libtool: 1: eval: base_compile+= -DOPERATION_add_1: not found
../libtool: 1: eval: base_compile+= -Wno-implicit-function-declaration: not found
../libtool: 1: eval: base_compile+= -fPIC: not found
../libtool: 1: eval: base_compile+= -c: not found
libtool: compile: you must specify a compilation command
libtool: compile: Try `libtool --help --mode=compile' for more information.
make[2]: *** [add_1.lo] Error 1
make[1]: *** [all-recursive] Error 1
make: *** [all] Error 2
```

**Comment by strega-nil** [comment]
That's very odd... that's the whole point of the `configure` script...

**Comment by strega-nil** [comment]
Apparently, I hadn't tested on Linux; @cenit could you try the new PR?

**Comment by zsxing99** [comment]
I had the same issue today. It worked well last Thursday.

OS: Ubuntu Linux 18.04

`vcpkg/buildtrees/mpir/build-x64-linux-dbg-err.log`
```
../libtool: 1581: ../libtool: preserve_args+= --tag CC: not found
../libtool: 1: eval: base_compile+= gcc: not found
../libtool: 1: eval: base_compile+= -DHAVE_CONFIG_H: not found
../libtool: 1: eval: base_compile+= -I.: not found
../libtool: 1581: ../libtool: preserve_args+= --tag CC: not found
../libtool: 1: eval: base_compile+= gcc: not found
../libtool: 1: eval: base_compile+= -DHAVE_CONFIG_H: not found
../libtool: 1: eval: base_compile+= -I.: not found
../libtool: 1: eval: base_compile+= -I.././../src/3.0.0-a3f571144a.clean/mpn: not found
../libtool: 1: eval: base_compile+= -I..: not found
../libtool: 1: eval: base_compile+= -D__GMP_WITHIN_GMP: not found
../libtool: 1: eval: base_compile+= -I.././../src/3.0.0-a3f571144a.clean/mpn: not found
../libtool: 1: eval: base_compile+= -I..: not found
../libtool: 1: eval: base_compile+= -D__GMP_WITHIN_GMP: not found
../libtool: 1: eval: base_compile+= -I.././../src/3.0.0-a3f571144a.clean: not found
../libtool: 1: eval: base_compile+= -DOPERATION_add: not found
../libtool: 1: eval: base_compile+= -Wno-implicit-function-declaration: not found
../libtool: 1: eval: base_compile+= -fPIC: not found
../libtool: 1: eval: base_compile+= -c: not found
libtool: compile: you must specify a compilation command
libtool: compile: Try `libtool --help --mode=compile' for more information.
make[2]: *** [add.lo] Error 1
make[2]: *** Waiting for unfinished jobs....
../libtool: 1: eval: base_compile+= -I.././../src/3.0.0-a3f571144a.clean: not found
../libtool: 1: eval: base_compile+= -DOPERATION_dummy1: not found
../libtool: 1: eval: base_compile+= -Wno-implicit-function-declaration: not found
../libtool: 1: eval: base_compile+= -fPIC: not found
../libtool: 1: eval: base_compile+= -c: not found
libtool: compile: you must specify a compilation command
libtool: compile: Try `libtool --help --mode=compile' for more information.
make[2]: *** [dummy1.lo] Error 1
make[1]: *** [all-recursive] Error 1
make: *** [all] Error 2
```
